### PR TITLE
[jenkins/docs] - added README for setting up a jenkins build slave

### PR DESCRIPTION
**Pull request by Memphiz**
… mac os (for building ios and osx with jenkins)

This is the HowTo for setting up a jenkins slave on macos. I did it exactly like that for the i7-1 slave.

@MartijnKaijser fyi

### CODE REVIEW

**Comment by Rechi** [comment]
Can you add the following after install ccache?
`Increase ccache max_size if enough disk space available`

If you want the exact location of the file that needs to be changed just tell me.

**Comment by Memphiz** [comment]
Ah right - totally forgot about that one ... can you tell me the file and the size? so i can add it to the readme (and to the 2 new slaves too)

**Comment by Rechi** [comment]
The file location is `/var/lib/jenkins/.ccache/ccache.conf` (`$HOME/.ccache/ccache.conf`) and I had set the content to `max_size = 15.0G` on the i5 nodes.

File: tools/buildsteps/jenkins_docs/README.mac [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+sudo chown -R jenkins:wheel /var/lib/jenkins
+# create a logging space
+sudo mkdir /var/log/jenkins
+sudo chown jenkins:wheel /var/log/jenkins
+
+3. mkdir /Users/Shared/jenkins
+
+4. sudo chown jenkins:wheel /Users/Shared/jenkins
+
+5. mkdir -p /Users/Shared/xbmc-depends/dSyms
```
Comment (Rechi) [inline tools/buildsteps/jenkins_docs/README.mac]: seems like `/dSyms` went in by accident
same in the next line

**Comment by Kwiboo** [comment]
You can use a script such as https://github.com/RasPlex/jenkins-slave-osx to simplify part of the jenkins slave setup. I used that script to configure my osx jenkins slaves for openpht in a matter of seconds.

**Comment by Memphiz** [comment]
@Kwiboo i know that script and i used part of it. I am not comfortable to run an "arbitrary" script on our jenkins nodes thats why i pulled the steps we need out of it and put those in the readme. So i know what is done to the slave.

File: tools/buildsteps/jenkins_docs/README.mac (hunk 1 above)
Comment (Memphiz) [inline tools/buildsteps/jenkins_docs/README.mac]: nope its not. Jenkins puts the generated debug info for each build into that directory when it exists. We use it mostly during releases when i need to symbolicate osx or ios crashlogs which don't have symbols when coming from a user.

Comment (Rechi) [inline tools/buildsteps/jenkins_docs/README.mac]: but jenkins must have permission for /Users/Shared/xbmc-depends
so at least it is wrong in the next line

**Comment by Memphiz** [comment]
added ccache cache size adaption to the readme and also set it on the new nodes... - also fixed the dSyms dir permissions in the Readme (thx for pointing it out @Rechi

**Comment by Memphiz** [comment]
squashed - i think this is good to go?

**Comment by Rechi** [comment]
looks correct

**Pull request merged**